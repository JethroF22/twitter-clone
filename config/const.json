{
  "actionStatusMessages": {
    "FAILED_MESSAGE": "Failed",
    "IN_PROGRESS_MESSAGE": "In Progress",
    "SUCCESS_MESSAGE": "Successful"
  },
  "actionTypes": {
    "auth": {
      "SET_USER_DETAILS": "SET_USER_DETAILS",
      "SET_TOKEN": "SET_TOKEN"
    },
    "error": {
      "SET_ERROR_MESSAGE": "SET_ERROR_MESSAGE"
    },
    "profile": {
      "SET_USER_PROFILE": "SET_USER_PROFILE",
      "VIEW_PROFILE": "VIEW_PROFILE"
    },
    "status": {
      "SET_ACTION_STATUS": "SET_ACTION_STATUS"
    },
    "tweet": {
      "SET_USER_TWEETS": "SET_USER_TWEETS",
      "UPDATE_USER_TWEETS": "UPDATE_USER_TWEETS"
    }
  },
  "errorTypes": {
    "AUTHENTICATION_ERROR": "Authentication Error",
    "AUTHORISATION_ERROR": "Authorisation Error",
    "DB_ERROR": "DB Error",
    "INVALID_REQUEST": "Invalid Request"
  },
  "errorMessages": {
    "CANT_BE_FOLLOWED": "Users cannot follow themselves",
    "DUPLICATE_EMAIL": "Duplicate email",
    "DUPLICATE_HANDLE": "Duplicate handle",
    "HAS_BEEN_LIKED": "User has already liked this tweet",
    "HAS_NOT_BEEN_LIKED": "User has yet to like this tweet",
    "HAS_BEEN_RETWEETED": "Cannot retweet the same tweet more than once",
    "HAS_BEEN_FOLLOWED": "Already following user",
    "HAS_NOT_BEEN_FOLLOWED": "User is not currently being followed",
    "INVALID_CREDENTIALS": "Invalid email/password combination",
    "TWEET_NOT_FOUND": "Tweet not found",
    "TWEETS_NOT_FOUND": "User has yet to tweet anything",
    "UNKNOWN_ERROR": "Unknown DB error",
    "UNAUTHORISED": "Unauthorised",
    "USER_NOT_FOUND": "User not found",
    "VALIDATION_ERROR": "Validation Error"
  }
}
