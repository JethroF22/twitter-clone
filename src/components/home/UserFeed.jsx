import React from 'react';
import { connect } from 'react-redux';

import TweetForm from '../forms/tweets/TweetForm';
import TweetList from '../lists/TweetList';

const UserFeed = props => (
  <div>
    <TweetForm />
    <TweetList id={props.id} />
  </div>
);

const mapStateToProps = state => ({
  id: state.auth._id
});

export default connect(mapStateToProps)(UserFeed);
