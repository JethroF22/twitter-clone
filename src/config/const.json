{
  "actionStatusMessages": {
    "SUCCESS_MESSAGE": "Successful",
    "IN_PROGRESS_MESSAGE": "In Progress",
    "FAILED_MESSAGE": "Failed"
  },
  "actionTypes": {
    "auth": {
      "SET_USER_DETAILS": "SET_USER_DETAILS",
      "SET_TOKEN": "SET_TOKEN"
    },
    "status": {
      "SET_ACTION_STATUS": "SET_ACTION_STATUS"
    },
    "error": {
      "SET_ERROR_MESSAGE": "SET_ERROR_MESSAGE"
    }
  }
}
