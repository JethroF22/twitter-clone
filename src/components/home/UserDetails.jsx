import React, { Component } from 'react';

import { actionStatusMessages } from '../../../config/const.json';

class UserDetails extends Component {
  render() {
    const profile = this.props.profile;
    return (
      <div>
        <img src={profile.coverPhoto} />
        <img src={profile.photo} />
        <h4>{profile.name}</h4>
        <p>{profile.handle}</p>
        <p>{this.props.tweets.length}</p>
        <p>{profile.followers.length}</p>
        <p>{profile.following.length}</p>
      </div>
    );
  }
}

export default UserDetails;
