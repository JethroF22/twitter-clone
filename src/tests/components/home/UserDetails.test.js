import React from 'react';
import { shallow } from 'enzyme';
import _ from 'lodash';

import { UserDetails } from '../../../components/home/UserDetails';
import { users } from '../../../../config/seed';
import { actionStatusMessages } from '../../../../config/const.json';
const { SUCCESS_MESSAGE, IN_PROGRESS_MESSAGE } = actionStatusMessages;

let getUserProfileSpy, id, name, user, wrapper;

describe('UserDetails', () => {
  beforeEach(() => {
    getUserProfileSpy = jest.fn().mockResolvedValueOnce(null);
    user = users[2];
    id = user._id;
    name = user.name;
    wrapper = shallow(
      <UserDetails
        id={id}
        name={name}
        tweets={[]}
        getUserProfile={getUserProfileSpy}
        actionStatus={SUCCESS_MESSAGE}
        profile={user}
      />
    );
  });

  test('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should dispatch `getUserProfile` on mount', () => {
    expect(getUserProfileSpy).toHaveBeenLastCalledWith(id, name);
  });

  test('should display message while `getUserProfile` in progress', () => {
    wrapper = shallow(
      <UserDetails
        id={id}
        name={name}
        tweets={[]}
        getUserProfile={getUserProfileSpy}
        actionStatus={IN_PROGRESS_MESSAGE}
      />
    );
    expect(wrapper.contains(<p>Loading ...</p>)).toBeTruthy();
  });
});
