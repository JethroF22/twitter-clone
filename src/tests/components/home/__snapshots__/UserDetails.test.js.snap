// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserDetails should render correctly 1`] = `
<div>
  <img
    src="https://images.pexels.com/photos/1130287/pexels-photo-1130287.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=350"
  />
  <img
    src="https://images.pexels.com/photos/1416822/pexels-photo-1416822.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=650&w=940"
  />
  <h4>
    Alexandra
  </h4>
  <p>
    QueenAlexa
  </p>
  <p>
    0
     tweets
  </p>
  <p>
    0
     followers
  </p>
  <p>
    0
     following
  </p>
</div>
`;
